### PR TITLE
chore: add DeployedJob selector method

### PR DESCRIPTION
**Pull request by bvtujo**
- chore: add DeployedJob method to DeploySelector and refactor filters to be job/svc agnostic.
- chore: fix mockgen
- test: add unit tests for DeployedJob()

This also fixes the makefile as far as I can tell. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/term/selector/selector.go [hunk 1]
```diff
@@ -534,9 +545,91 @@ func (s *CFTaskSelector) Task(msg, help string, opts ...GetDeployedTaskOpts) (st
 	return choice, nil
 }
 
+// DeployedJob has the user select a deployed job. Callers can provide either a particular environment,
+//// a particular service to filter on, or both.
```
Comment (paragbhingre) [inline internal/pkg/term/selector/selector.go]:
```suggestion
// a particular service to filter on, or both.
```

File: internal/pkg/term/selector/selector.go [hunk 2]
```diff
@@ -484,6 +485,16 @@ func (s *DeployedService) String() string {
 	return fmt.Sprintf("%s (%s)", s.Svc, s.Env)
 }
 
+// DeployedJob contains the job name and environment name of the deployed job.
+type DeployedJob struct {
+	Job string
+	Env string
+}
+
+func (j *DeployedJob) String() string {
```
Comment (paragbhingre) [inline internal/pkg/term/selector/selector.go:499]: can we put comment to this public function?

File: internal/pkg/term/selector/selector.go (hunk 1 above)
Comment (paragbhingre) [inline internal/pkg/term/selector/selector.go]:
```suggestion
// a particular job to filter on, or both.
```

**Review by paragbhingre — APPROVED** [review]
LGTM🎉 just few tiny nits

File: internal/pkg/term/selector/selector.go [hunk 3]
```diff
@@ -534,9 +545,91 @@ func (s *CFTaskSelector) Task(msg, help string, opts ...GetDeployedTaskOpts) (st
 	return choice, nil
 }
 
+// DeployedJob has the user select a deployed job. Callers can provide either a particular environment,
+//// a particular service to filter on, or both.
+func (s *DeploySelector) DeployedJob(msg, help string, app string, opts ...GetDeployedWorkloadOpts) (*DeployedJob, error) {
```
Comment (iamhopaul123) [inline internal/pkg/term/selector/selector.go:567]: It looks like this method has a lot of duplication as `DeployedService`. I wonder if we should instead add a `deployedWorkloads` and having both `DeployedServices` and `DeployedJobs` to use it.

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**